File: 3 - largest prime factor/solution.py
#!/usr/bin/env python3

def main():
    n = 600_851_475_143
    i = 2
    while i ** 2 < n:
        while not n % i:
            n //= i
        i += 1
    print(n)


if __name__ == '__main__':
    main()

File: 3 - largest prime factor/solution.jl
#!/usr/bin/env julia

import Pkg
Pkg.add("Primes")
import Primes

n = 600_851_475_143
println(iterate(n ÷ i for i in 1:n if n % i == 0 && Primes.isprime(n ÷ i))[1])
